HopsanCore/src/ComponentSystem.cpp

```diff
@@ -3381,6 +3383,18 @@ void ComponentSystem::simulate(const double stopT)
     }
 }
 
+bool ComponentSystem::startRealtimeSimulation(double realTimeFactor)
+{
+#if (__cplusplus >= 201103L)
+    std::thread rtThread(simWholeSystemInRealtime, realTimeFactor, &mStopSimulation, &mTime, mTimestep, mComponentSignalptrs, mComponentCptrs, mComponentQptrs);
+    rtThread.detach();
```

Comment:
Minor risk, but maybe up to application, when we start a detached simulation like this we do not know if it is running or what it is doing. We could still start ordinary simulation or another copy of detached rt simulation. That would use the same model at the same time.
And what happens if a thread locks up, we can not be sure that it is complete. (Except look at end time) But I dont know if it would make much difference if we had a handle to the thread, we cant do anything with it anyway,
So I guess its OK

Reply:
Similar problems exist with normal simulations in nonblocking mode. If a component e.g. gets stuck in a loop you need to restart Hopsan, because the simualte mutex will never be unlocked.

I will move the start and stop functions to ModelWidget though, so that we can at least lock and unlock the simualte mutex to avoid starting a second simulation while the first is still running.